modules/standard/GpuDiagnostics.chpl

```diff
@@ -56,6 +56,70 @@ module GpuDiagnostics
     var device_to_device: uint(64);
   };
 
+  /*
+   * Utility function to assert GPU diagnostic values. First 4 formals have the
+   * same names as the fields in chpl_gpuDiagnostics type. `_um` and `_aod`
+   * versions of `kernel_launch` are for convenience where we expect different
+   * number of launches in different memory modes
+   */
+  @chpldoc.nodoc
+  proc assertGpuDiags(kernel_launch=-1,
+                      host_to_device=-1,
+                      device_to_host=-1,
+                      device_to_device=-1,
+                      kernel_launch_um=-1,
+                      kernel_launch_aod=-1) {
+
+    /* doesn't do any comparison for expected<0. If compared, and
+       expected != actual, prints out a message and returns false. Otherwise,
+       returns true quietly. */
+    proc compare(expected, actual, name) {
+      const res = (expected<0) || (actual == expected);
+
+      if res == false {
+        writef("Unexpected GPU Diagnostic: observed %i %s instead of %i\n",
+               actual, name, expected);
+      }
+
+      return res;
+    }
+
+    use ChplConfig;
+
+    param isUm = CHPL_GPU_MEM_STRATEGY == "unified_memory";
+
+    const expectedLaunch;
+    if kernel_launch >= 0 {
+      if kernel_launch_um >= 0 {
+        writeln("Warning: kernel_launch_um argument is ignored because ",
```

Comment:
Any reason not to just make this a runtime error?

Reply:
I thought this would be less annoying, but can switch to halt/assert. No deep reason.

Reply:
(unless I'm missing something) I'd say anyone doing this is most likely making a mistake so it should be annoying/halting. This could be lost in the noise for a test that filters its output through a `.prediff` file or something.

Reply:
OK, I am convinced. I'll make this more loud.